Dockerfile.py38

```diff
@@ -0,0 +1,15 @@
+FROM python:3.8-slim-buster
+
+MAINTAINER Flyte Team <users@flyte.org>
+LABEL org.opencontainers.image.source https://github.com/flyteorg/flytekit
+
+ARG VERSION
+ARG DOCKER_IMAGE
```

Comment:
remind me how this works?  so even though `DOCKER_IMAGE` is defined on line 7, because it's not used until the end, it won't break caching right?  Like we're not going to rebuild the aws/gsutil layers every time the version changes right?  Doesn't really matter since this is a smaller image, just wondering.

Reply:
Yes i made the changes